orderBook.js

```diff
@@ -1,5 +1,46 @@
+function findMatch(existingBook, incomingOrder) {
+  return existingBook.filter(bookItem => (bookItem.type !== incomingOrder.type)
+    && !((bookItem.price !== incomingOrder.price) && (bookItem.quantity !== incomingOrder.quantity)))
+}
+
+function filterMatch(existingBook, incomingOrder) {
+  return existingBook.filter(bookItem => (bookItem.type === incomingOrder.type)
+    && (bookItem.quantity <= incomingOrder.quantity)
+    && (bookItem.price === incomingOrder.price))
+}
+
+function priceDecider(existingPrice, incomingPrice) {
+  const priceIndex = existingPrice - incomingPrice
+  if (priceIndex >= 0) {
+    return true
+  }
+}
+
 function reconcileOrder(existingBook, incomingOrder) {
-  throw new Error('Not Implemented Yet')
+  const filteredBook = findMatch(existingBook, incomingOrder)
```

Comment:
This naming is funky. You have a `filterMatch` function but are calling the return value of `findMatch` `filteredBook`. Makes it a little tougher to track what's happening.